### PR TITLE
Store secrets firebase

### PR DESCRIPTION
**Pull request by juankysoriano**
# Short Description

Introduces a script that fetchs the api secrets and stores it in an untracked file by git (`.secrets_env`)

This will grant the app access to the API_KEY value so that api calls can succeed

# What changes in the code?

Adds the script
Modifies the remote data source so that it will eventually read the api key from the `.secrets_env` file (`NEWS_API_KEY`) variable

# Did you test?

- [x] Yes
- [ ] No

*Note*: Explain why you didn't test if that's the case

### CODE REVIEW

**Comment by juankysoriano** [comment]
@Diogo00dev looks like the tests are failing because not being able to find the file `.secrets_env` which means I have the chance to rework the github action to fetch the secrets as part of this PR, I will close and reopen once that is done

File: .github/workflows/news-check-pr.yml [hunk 1]
```diff
@@ -9,19 +9,40 @@ jobs:
     runs-on: ubuntu-latest
     name: Run Test and Static Analysis
     steps:
-      - uses: actions/checkout@v3     
-      - uses: subosito/flutter-action@v2
+      - uses: actions/checkout@v3
+
+      - id: 'auth'
```
Comment (juankysoriano) [inline .github/workflows/news-check-pr.yml:14]: Step 1 is being able to authenticate with Firebase

File: .github/workflows/news-check-pr.yml [hunk 2]
```diff
@@ -9,19 +9,40 @@ jobs:
...
+        uses: google-github-actions/auth@v0
+        with:
+          credentials_json: '${{ secrets.gcp_credentials }}'
+
+      - id: 'secrets'
```
Comment (juankysoriano) [inline .github/workflows/news-check-pr.yml:19]: Step 2 is to get the api key from secrets

File: .github/workflows/news-check-pr.yml [hunk 3]
```diff
@@ -9,19 +9,40 @@ jobs:
...
+        uses: google-github-actions/get-secretmanager-secrets@v0
+        with:
+          secrets: |-
+            news-api-key:projects/859887479109/secrets/NEWS_API_KEY/versions/1
+
+      - name: Creates local file with API key
```
Comment (juankysoriano) [inline .github/workflows/news-check-pr.yml:25]: Finally we create the local file that contains the api key

**Comment by juankysoriano** [comment]
@Diogo00dev this is ready to be merged, once it is please reopen #17 and review

Unfortunately we cannot run the script I created to be run by devs, `fetch-secrets.sh` from GitHub Actions, here things are slightly different.

I described the steps that are required as part of the workflow on the `.workflow` file

**Comment by Diogo00dev** [comment]
LGTM, thanks for the explanation in each step 💪

**Pull request merged**